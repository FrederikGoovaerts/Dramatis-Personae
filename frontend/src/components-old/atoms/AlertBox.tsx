import { Box, Paper, Typography } from '@material-ui/core';
import Warning from '@material-ui/icons/Warning';
import * as React from 'react';

import { grayLight } from '../../assets/Colors';

interface Props {
    text: string;
}

export const AlertBox = (props: Props) => (
    <Paper>
        <Box
            padding="0.5em"
            display="flex"
            flexDirection="row"
            borderRadius="0.2em"
            alignItems="center"
            bgcolor={grayLight}
        >
            <Box margin="0.5em">
                <Warning />
            </Box>
            <Typography>{props.text}</Typography>
        </Box>
    </Paper>
);
