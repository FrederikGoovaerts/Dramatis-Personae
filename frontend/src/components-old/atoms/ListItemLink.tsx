import ListItem from '@material-ui/core/ListItem';
import * as React from 'react';
import { Link } from 'react-router-dom';

export interface Props {
    to: string;
    children: React.ReactNode;
    selected?: boolean;
}

export const ListItemLink = (props: Props) => <ListItem selected={props.selected} button component={Link} {...props} />;
