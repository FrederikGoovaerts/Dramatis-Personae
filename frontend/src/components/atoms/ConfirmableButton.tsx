import { Button } from '@material-ui/core';
import React, { useState } from 'react';

type ButtonState = 'DEFAULT' | 'CLICKED' | 'DELETING';

interface Props {
    onConfirm: () => void;
    defaultText: string;
    confirmedText: string;
    disabled?: boolean;
}

export const ConfirmableButton = (props: Props) => {
    const [state, setState] = useState<ButtonState>('DEFAULT');

    const click = () => {
        if (state === 'DEFAULT') {
            setState('CLICKED');
        } else if (state === 'CLICKED') {
            setState('DELETING');
            props.onConfirm();
        }
    };

    const variant = state === 'DEFAULT' ? 'outlined' : 'contained';
    const disabled = state === 'DELETING' || props.disabled;
    const contents = state === 'DEFAULT' ? props.defaultText : state === 'CLICKED' ? 'Confirm' : props.confirmedText;

    return (
        <Button variant={variant} color="secondary" onClick={click} disabled={disabled}>
            {contents}
        </Button>
    );
};

interface PredefinedButtonProps {
    onConfirm: () => void;
    disabled?: boolean;
}

export const DeleteButton = (props: PredefinedButtonProps) => (
    <ConfirmableButton
        onConfirm={props.onConfirm}
        defaultText="Delete"
        confirmedText="Deleting..."
        disabled={props.disabled}
    />
);

export const MergeButton = (props: PredefinedButtonProps) => (
    <ConfirmableButton
        onConfirm={props.onConfirm}
        defaultText="Merge into"
        confirmedText="Merging..."
        disabled={props.disabled}
    />
);
