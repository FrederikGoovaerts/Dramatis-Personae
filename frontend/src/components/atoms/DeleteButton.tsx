import * as React from 'react';
import { Button } from '@material-ui/core';

type ButtonState = 'DEFAULT' | 'CLICKED' | 'DELETING';

interface Props {
    onConfirm: () => void;
    defaultText: string;
    confirmedText: string;
}

interface State {
    state: ButtonState;
}

export class ConfirmableButton extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = { state: 'DEFAULT' };
    }

    private click = () => {
        this.setState({ state: 'CLICKED' });
    };

    private confirm = () => {
        this.setState({ state: 'DELETING' });
        this.props.onConfirm();
    };

    render() {
        if (this.state.state === 'DEFAULT') {
            return (
                <Button variant="outlined" color="secondary" onClick={this.click}>
                    {this.props.defaultText}
                </Button>
            );
        }
        if (this.state.state === 'CLICKED') {
            return (
                <Button variant="contained" color="secondary" onClick={this.confirm}>
                    Confirm
                </Button>
            );
        }
        return (
            <Button variant="contained" color="secondary" disabled>
                {this.props.confirmedText}
            </Button>
        );
    }
}

interface DeleteButtonProps {
    onConfirm: () => void;
}

export const DeleteButton = (props: DeleteButtonProps) => (
    <ConfirmableButton onConfirm={props.onConfirm} defaultText="Delete" confirmedText="Deleting..." />
);
