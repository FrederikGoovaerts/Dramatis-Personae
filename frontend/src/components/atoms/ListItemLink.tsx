import ListItem from '@material-ui/core/ListItem';
import * as React from 'react';
import { Link } from 'react-router-dom';

export interface Props {
    to: string;
    children: React.ReactNode;
    selected?: boolean;
}

export function ListItemLink(props: Props) {
    return <ListItem selected={props.selected} button component={Link} {...props} />;
}
