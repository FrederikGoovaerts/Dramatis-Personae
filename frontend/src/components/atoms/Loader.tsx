import { Box, CircularProgress } from '@chakra-ui/react';
import React from 'react';

export const Loader = () => (
    <Box display="flex" justifyContent="center" marginTop="2em" marginBottom="2em">
        <CircularProgress isIndeterminate color="teal.500" />
    </Box>
);
