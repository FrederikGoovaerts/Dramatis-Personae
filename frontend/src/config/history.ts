import { createBrowserHistory } from 'history';
import { hosting } from './constants';

export const history = createBrowserHistory({
    basename: hosting.BASE_URL
});
