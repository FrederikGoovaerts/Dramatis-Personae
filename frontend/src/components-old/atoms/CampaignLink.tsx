import { makeStyles, Paper, Typography } from '@material-ui/core';
import React from 'react';
import { Link } from 'react-router-dom';

import { routes } from '../../config/constants';

interface Props {
    id: string;
    name: string;
    ownerName: string;
}

const useStyles = makeStyles({
    link: {
        display: 'flex',
        margin: '1em',
        flexDirection: 'column'
    },
    paper: {
        flex: 1,
        padding: '2em',
        maxWidth: '10em'
    }
});

export const CampaignLink = (props: Props) => {
    const classes = useStyles();
    return (
        <Link to={`${routes.campaign.path}${props.id}`} key={props.id} className={classes.link}>
            <Paper className={classes.paper}>
                <Typography variant="h5" gutterBottom>
                    {props.name}
                </Typography>
                <Typography variant="subtitle1">Run by {props.ownerName}</Typography>
            </Paper>
        </Link>
    );
};
