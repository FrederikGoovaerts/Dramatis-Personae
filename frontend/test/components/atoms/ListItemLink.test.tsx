import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import { create } from 'react-test-renderer';

import { ListItemLink } from '../../../src/components/atoms/ListItemLink';

describe('The ListItemLink', () => {
    it('should render correctly', () => {
        const component = create(
            <Router history={createMemoryHistory()}>
                <ListItemLink to="destination">contents</ListItemLink>
            </Router>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
