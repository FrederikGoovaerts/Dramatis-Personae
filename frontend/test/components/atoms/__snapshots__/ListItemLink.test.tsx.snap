// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ListItemLink should render correctly 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  href="/destination"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  contents
</a>
`;
